nuxeo-vision-core/src/main/java/org/nuxeo/vision/core/service/VisionProviderDescriptor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Michael Vachette
+ */
+package org.nuxeo.vision.core.service;
+
+import org.nuxeo.common.xmap.annotation.XNode;
+import org.nuxeo.common.xmap.annotation.XNodeMap;
+import org.nuxeo.common.xmap.annotation.XObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @since 9.1
+ */
+@XObject("provider")
+public class VisionProviderDescriptor {
+
+    @XNode("@name")
+    protected String providerName;
+
+    @XNode("@class")
+    protected Class className;
+
+    @XNodeMap(value = "parameters/parameter", key = "@name", type = HashMap.class, componentType = String.class)
+    protected Map<String, String> parameters = new HashMap<>();
+
+    public Class getClassName() {
```

Comment:
This returns a `Class` (actually should be a `Class<?>`, and the field too), so should be called `getKlass` (as `getClass` is reserved).